lib/transaction.js

```diff
@@ -74,13 +74,7 @@ class Transaction extends EventEmitter {
           return makeTransactor(this, connection, trxClient);
         })
         .then((transactor) => {
-          if (this.initPromise) {
-            transactor.executionPromise = executionPromise.catch((err) => {
-              throw err;
-            });
-          } else {
-            transactor.executionPromise = executionPromise;
-          }
+          transactor.executionPromise = executionPromise;
```

Comment:
@maximelkin :  I'm fairly certain that this change is the one that fixed the race condition.

Your PR (#3665) included this change as well; however, it relocated the `.catch((err)=> {throw err})` logic to a different part of the code.  In contrast, this PR just removes that logic entirely.

Reply:
This catch added in #3328, so I am unsure about this changes